File: lib/limelight/builtin/players/check_box.rb
#- Copyright © 2008-2010 8th Light, Inc. All Rights Reserved.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

on_cast do
  check_box = Limelight::UI::Model::Inputs::CheckBoxPanel.new
  panel.add(check_box)
  @check_box = check_box
end

attr_accessor :check_box #:nodoc:

# Will place or remove a check mark in the check box.
#
def checked=(value)
  check_box.selected = value
end

# Returns true if the check box is checked.
#
def checked
  return check_box.selected?
end

alias :checked? :checked

File: src/limelight/builtin/players/CheckBox.java
package limelight.builtin.players;

import limelight.ui.events.panel.PanelEvent;
import limelight.ui.model.PropPanel;
import limelight.ui.model.inputs.CheckBoxPanel;

public class CheckBox
{
  private PropPanel propPanel;
  private CheckBoxPanel checkBoxPanel;

  public void install(PanelEvent event)
  {
    checkBoxPanel = new CheckBoxPanel();
    propPanel = (PropPanel)event.getRecipient();
    propPanel.add(checkBoxPanel);
  }

  public PropPanel getPropPanel()
  {
    return propPanel;
  }

  public CheckBoxPanel getCheckBoxPanel()
  {
    return checkBoxPanel;
  }

  public boolean isChecked()
  {
    return checkBoxPanel.isSelected();
  }

  public boolean isSelected()
  {
    return checkBoxPanel.isSelected();
  }

  public void setChecked(boolean b)
  {
    checkBoxPanel.setSelected(b);
  }

  public void setSelected(boolean b)
  {
    checkBoxPanel.setSelected(b);
  }
}
